.sandstorm/get-sandstorm-id.py

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/python3
+from http.server import BaseHTTPRequestHandler, HTTPServer
+import os, tempfile, sys
+
+os.chdir(os.path.split(__file__)[0])
+
+[_, serverPort, reloadDelay, envOutputPath] = sys.argv
+serverPort = int(serverPort)
+
+hostName = "0.0.0.0"
+
+class MyServer(BaseHTTPRequestHandler):
+    def write_id(self, sandstorm_id):
+        dir_path = os.path.split(envOutputPath)[0]
+        f = tempfile.NamedTemporaryFile(mode='w', dir=dir_path, delete=False)
+        f.write(sandstorm_id)
+        f.close()
+        os.rename(f.name, envOutputPath)
+
+    def do_GET(self):
+        self.write_id(self.headers['X-Sandstorm-User-Id'])
+        self.send_response(200)
+        self.send_header("Content-type", "text/html")
+        self.end_headers()
+
+        # reloadDelay ms "hopefully" is enough time for the next server to start
```

Comment:
This could be risky with the migration. I'm not sure how long it'll take. Sandstorm might be forgiving on the first request and let you do a long migration, but this makes an extra request to get the sandstorm ID. If you set it too low, the next server won't be there when it reloads the page. I wonder if Caddy could handle it smoothly though.